src/Features/Core/Portable/CommentSelection/ICommentSelectionService.cs

```diff
@@ -0,0 +1,16 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Immutable;
+using System.Threading;
+using Microsoft.CodeAnalysis.Host;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Microsoft.CodeAnalysis.CommentSelection
+{
+    internal interface ICommentSelectionService : ILanguageService
+    {
+        CommentSelectionInfo GetInfo(SourceText sourceText, TextSpan textSpan);
```

Comment:
We give you the text we're going to comment/uncomment and the span of the user selection.  You probably only need to look at the start of the span and decide what sort of comment info you want (i.e. if you're inside TS code, or if you're inside JSX code).  

Note: i have no idea what you want to do if the user makes a selection that *crosses* the languages.  :)